x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credential_form.tsx

```diff
@@ -240,56 +247,77 @@ const GcpInputVarFields = ({
   fields: Array<GcpFields[keyof GcpFields] & { value: string; id: string }>;
   onChange: (key: string, value: string) => void;
 }) => {
-  const [credentialOption, setCredentialOption] = useState('Credentials File');
-  const targetFieldName = (id: string) => {
+  const getFieldById = (id: keyof GcpInputFields['fields']) => {
     return fields.find((element) => element.id === id);
   };
+  const projectIdFields = getFieldById('project_id');
+  const credentialsTypeFields = getFieldById('credentials_type');
+  const credentialFilesFields = getFieldById('credentials_file');
+  const credentialJSONFields = getFieldById('credentials_json');
+  const credentialsTypeField = {
+    id: 'credentials_type',
+    label: 'Credentials Type',
+    type: 'text',
+    value: credentialsTypeFields || credentialOptionsList[0].value,
+  };
```

Comment:
having both `credentialsTypeField` and `credentialsTypeFields` is confusing, and id, label and type are not really being used here, so I think it's better to remove `credentialsTypeField` and update `credentialsTypeFields`  to the following:

```js
const credentialsTypeFields = getFieldById('credentials_type') || credentialOptionsList[0];
```

Then the select will have the selected value or initialize with the default.